### PR TITLE
chore: bump dependencies

### PR DESCRIPTION
**Pull request by Unix4ever**
Fixes: https://github.com/talos-systems/talos/pull/3954 https://github.com/talos-systems/talos/pull/3955 https://github.com/talos-systems/talos/pull/3956 https://github.com/talos-systems/talos/pull/3957 https://github.com/talos-systems/talos/pull/3958 https://github.com/talos-systems/talos/pull/3959 https://github.com/talos-systems/talos/pull/3960 https://github.com/talos-systems/talos/pull/3961 https://github.com/talos-systems/talos/pull/3962 https://github.com/talos-systems/talos/pull/3963 https://github.com/talos-systems/talos/pull/3964

And update kubelet to 1.21.3.

Signed-off-by: Artem Chernyshev <artem.0xD2@gmail.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -105,13 +104,12 @@ require (
...
-	k8s.io/cri-api v0.21.2
-	k8s.io/kubectl v0.21.2
-	k8s.io/kubelet v0.21.2
+	k8s.io/api v0.21.3
+	k8s.io/apimachinery v0.21.3
+	k8s.io/apiserver v0.21.3 // indirect; use the same version as other k8s deps
+	k8s.io/client-go v0.21.3
+	k8s.io/cri-api v0.21.3
+	k8s.io/kubectl v0.21.3
+	k8s.io/kubelet v0.21.3
```
Comment (AlekSi) [inline go.mod:113]: It will be updated to 1.22-beta.2 in https://github.com/talos-systems/talos/pull/3935. Please wait a bit

Comment (AlekSi) [reply go.mod:113]: Also, it was already backported into 0.11.2: https://github.com/talos-systems/talos/commit/965558787734c9001e89334788c0588a580339d6

Comment (AlekSi) [reply go.mod:113]: Ok, let's merge that one, and I will rebase (and fix) #3935 after that

Comment (Unix4ever) [reply go.mod:113]: yeah... I guess I get it now. The plan was to go straight to 1.22 in master.

Comment (AlekSi) [reply go.mod:113]: Yeah, it was, but 1.22 doesn't work yet :)

**Review by AlekSi — APPROVED** [review]

**Pull request merged**